lib/private/L10N/Factory.php

```diff
@@ -490,10 +490,14 @@ private function getLanguageFromRequest(?string $app = null): string {
 				[$preferred_language] = explode(';', $preference);
 				$preferred_language = str_replace('-', '_', $preferred_language);
 
+				$preferred_language_parts = explode('_', $preferred_language);
 				foreach ($available as $available_language) {
 					if ($preferred_language === strtolower($available_language)) {
 						return $this->respectDefaultLanguage($app, $available_language);
 					}
+					if ($preferred_language_parts[0].'_'.$preferred_language_parts[count($preferred_language_parts) - 1] === strtolower($available_language)) {
```

Comment:
```suggestion
					if ($preferred_language_parts[0].'_'.end($preferred_language_parts) === strtolower($available_language)) {
```

Maybe this is cleaner? (Not sure I am not an PHP coding style expert).